examples/jsm/loaders/KTX2Loader.js

```diff
@@ -272,7 +272,24 @@ class KTX2Loader extends Loader {
 
 		if ( container.vkFormat !== VK_FORMAT_UNDEFINED ) {
 
-			return createDataTexture( container );
+			const mipmaps = [];
+			const pendings = [];
+
+			for ( let levelIndex = 0; levelIndex < container.levels.length; levelIndex ++ ) {
+
+				pendings.push( createDataTexture( container, levelIndex ).then( function ( dataTexture ) {
+
+					mipmaps[ levelIndex ] = dataTexture;
```

Comment:
I don't think it's safe to reference `levelIndex` within this callback – it may have changed before the async function completes.

Reply:
That usage is safe. It's a `let`-based index (block-scoped local variable).

Reply:
Ah you're right, thanks. Bad memories of `var` and Internet Explorer.